docs/proposals/error-handling-guideline.md

```diff
@@ -0,0 +1,225 @@
+# ORAS CLI Error Handling and Message Guidelines
+
+This document aims to provide the guidelines for ORAS contributors to improve existing error messages and error handling method as well as the new error output format. It will also provide recommendations for ORAS CLI contributors for how to write friendly and standard error messages and avoid inconsistent error handling & messages.
+
+## General guiding principles
+
+A clear and actionable error message is very important when raising an error, so make sure your error message describes clearly what the error is and tells users what they need to do if possible.
+
+First and foremost, make the error messages descriptive and informative. Error messages are expected to be helpful to troubleshoot where the user has done something wrong and the program is guiding them in the right direction. A great error message is recommended to contain the following:
+
+- Error code
+- Error title
```

Comment:
What distinguishes a title from a description? Perhaps we can provide greater clarity. While both the title and description may not always be informative, we can emphasize that they should convey what happened and why it happened.